docs/ai/conceptual/agents.md

```diff
@@ -0,0 +1,86 @@
+---
+title: "Let Agents and Copilots Bring Functionality to Your App"
+description: "Learn how agents and copilots intelligently extend the functionality of LLMs to efficiently meet user goals in .NET."
+author: catbutler
+ms.topic: concept-article #Don't change.
+ms.date: 04/15/2024
+
+#customer intent: As a .NET developer, I want to understand how agents and copilots extend the functionality of AI apps, so that my apps can handle any type of content and meet more user goals.
+
+---
+
+# How agents and copilots work with LLMs
+
+Agents and copilots both extend an LLM's capabilities by intelligently invoking external functionality, such as sending an email.
+
+- **An agent** is an artificial intelligence that can answer questions and automate processes for users. Agents can determine which functions will meet a user's goal, and then call those functions on the user's behalf.
+- **A copilot** is a type of agent that works side-by-side with a user. Unlike an agent, a copilot isn't fully automated&mdash;it relies on user interaction. A copilot can help a user complete a task by providing suggestions and recommendations.
+
+For example, suppose you're building an email chat helper app. Along with the LLM, you'll also need a way to perform email-related actions, as well as functions for searching, summarizing, determining intent, and the like. Creating the plugins is only half the battle: you still need to invoke the right functions at the right time, a process with many opportunities for errors and inefficiencies. An agent can handle it better.
```

Comment:
> Creating the plugins

It's not clear what these plugins would be, I'm assuming they would be the plugins for the LLM to handle email-related actions. If so, I'd mention them by name when you describe adding functionality to the LLM.

Reply:
Good call - I'll list the plugins.

Reply:
:+1: